lucetc/src/function.rs

```diff
@@ -431,9 +470,24 @@ impl<'a> FuncEnvironment for FuncInfo<'a> {
             .expect("function indices are valid");
         let func_decl = self.module_decls.get_func(unique_index).unwrap();
         let signature = func.import_signature(func_decl.signature.clone());
+
+        // if we're setting up a function ref for a call to an imported function, we'll need a
+        // trampoline to check the stack first. in that case, return the trampoline function
+        // instead.
         let colocated = !func_decl.imported();
+        let name = if colocated {
```

Comment:
I also think this might be not _quite_ right, Cranelift docs suggest that guest functions can be `!colocated` if the module is sufficiently large (f.ex larger than 128mb of code). We would then insert a trampoline between calls internal to the guest, and then have guests fail due to not having enough stack for a hostcall when not making a hostcall at all.

I haven't _actually seen_ guest functions that are `!colocated`, and I'm not sure if there's a better way to select imported functions here. Even guest functions were erroneously picked here, it would work (with an unintentional penalty of extra function call) so long as stack is available - the error mode in case this _is_ wrong is not _too_ bad.